packages/@aws-cdk/cloudformation-include/test/valid-templates.test.ts

```diff
@@ -444,10 +444,64 @@ describe('CDK Include', () => {
     });
   });
 
-  test("throws an exception when encountering a Resource type it doesn't recognize", () => {
+  test('can include a template with a custom resource that only defines a type', () => {
+    includeTestTemplate(stack, 'non-existent-resource-type.json');
+    expect(stack).toHaveResourceLike('AWS::MyService::Custom');
+  });
+
+  test('can include a template with a custom resource that has custom properties', () => {
+    includeTestTemplate(stack, 'custom-resource-properties.json');
+    expect(stack).toHaveResourceLike('AWS::MyService::Custom', {
+      "CustomProp": "CustomValue",
+    });
+  });
+
+  test('can include a template with a custom resource that uses cloudformation functions', () => {
+    includeTestTemplate(stack, 'custom-resource-with-functions.json');
+    expect(stack).toHaveResourceLike('AWS::MyService::Custom', {
+      "CustomStringProp": {
+        "Ref": "AWS::NoValue",
+      },
+    });
+  });
+
+  test('can include a template with a custom resource that uses attributes', () => {
+    includeTestTemplate(stack, 'custom-resource-with-attributes.json');
+    expect(stack).toHaveResourceLike('AWS::MyService::Custom', {
+      "Metadata": {
+        "Object1": "Value1",
+        "Object2": "Value2",
+      },
+      "CreationPolicy": {
+        "AutoScalingCreationPolicy": {
+          "MinSuccessfulInstancesPercent": 90,
+        },
+      },
+      "DeletionPolicy": "Retain",
+      "DependsOn": [ "CustomResource" ],
+    }, ResourcePart.CompleteDefinition);
+
+    expect(stack).toHaveResourceLike('AWS::MyService::AnotherCustom', {
+      "UpdatePolicy": {
+        "AutoScalingReplacingUpdate": {
+          "WillReplace": "false",
+        },
+      },
+      "UpdateReplacePolicy": "Retain",
+    }, ResourcePart.CompleteDefinition);
+  });
+
+  test('can ingest a custom resource that uses conditions', () => {
+    const cfnTempalte = includeTestTemplate(stack, 'custom-resource-with-condition.json');
+    const alwaysFalseCondition = cfnTempalte.getCondition('AlwaysFalseCond');
+
+    expect(cfnTempalte.getResource('CustomResource').cfnOptions.condition).toBe(alwaysFalseCondition);
+  });
+
+  test('throws an exception when an undefined condition is used', () => {
```

Comment:
```suggestion
  test("throws an exception when a custom resource uses a Condition attribute that doesn't exist in the template", () => {
```

I feel like I've made this comment many times before.

`undefined condition` means something completely different in this context (I would expect a test with that name  to test something like `"Condtion": null` in the template).

Reply:
Oh, I see...my bad, I've been misunderstanding the meaning of `undefined` in this context.